### PR TITLE
Mssql improvements and readme instructions

### PR DESCRIPTION
**Pull request by anthony2261**

### CODE REVIEW

File: backend/dataline/models/connection/schema.py [hunk 1]
```diff
@@ -112,11 +112,13 @@ def validate_dsn(value: str) -> str:
 
     # Simpler way to connect to postgres even though officially deprecated
     # This mirrors psql which is a very common way to connect to postgres
-    if value.startswith("postgres") and not value.startswith("postgresql"):
+    if value.startswith("postgres://"):
```
Comment (RamiAwar) [inline backend/dataline/models/connection/schema.py:115]: This change will now fail on valid strings that specify a valid driver.

We should keep it as it was, but remove the driver perhaps.

File: README.md [hunk 2]
```diff
@@ -162,6 +163,10 @@ Go through the following checklist to explore DataLine's features!
 - [ ] To really challenge it, ask a question that is answered with multiple results (charts, tables, etc.) - [example](https://www.youtube.com/watch?v=6ouSok9bOVo)
 - [ ] Add a profile picture
 
+### Supported databases
+
+See [instructions file](./dsn-instructions.md) for more details.
```
Comment (RamiAwar) [inline README.md:168]: We should say something about which DBs are supported here

File: backend/dataline/services/llm_flow/nodes.py [hunk 3]
```diff
@@ -84,6 +84,10 @@ def run(cls, state: QueryGraphState) -> QueryGraphStateUpdate:
 
         output_messages: list[BaseMessage] = []
         results: list[QueryResultSchema] = []
+        if len(last_message.tool_calls) == 1 and last_message.tool_calls[0]["name"] == "multi_tool_use.parallel":
```
Comment (RamiAwar) [inline backend/dataline/services/llm_flow/nodes.py:87]: noice

File: backend/dataline/services/llm_flow/nodes.py [hunk 4]
```diff
@@ -103,6 +107,42 @@ def run(cls, state: QueryGraphState) -> QueryGraphStateUpdate:
         # We return a list, because this will get added to the existing list
         return state_update(messages=output_messages, results=results)
 
+    @staticmethod
+    def fix_openai_multi_tool_use_bug(buggy_tool_call: ToolCall) -> list[ToolCall]:
+        """
```
Comment (RamiAwar) [inline backend/dataline/services/llm_flow/nodes.py:112]: This docstring is a little confusing, could use some context at the beginning saying "This is the structure. I stared at it for a bit before realizing its a docstring lol.

File: backend/dataline/models/connection/schema.py (hunk 1 above)
Comment (anthony2261) [reply backend/dataline/models/connection/schema.py:115]: Can you give me an example of a valid string where this would fail?

Comment (RamiAwar) [reply backend/dataline/models/connection/schema.py]: Just added a commit to explain what I mean, will test it out locally before merging

Comment (RamiAwar) [reply backend/dataline/models/connection/schema.py]: Would be cool to just remove their DSN instead.
An example would be if they did: postgresql+whatever://etc.
Your changes would then fail to detect starts with 'postgres://'

image

Comment (anthony2261) [reply backend/dataline/models/connection/schema.py]: Yep, with driver detection and removal it makes sense

**Review by RamiAwar — APPROVED** [review]

**Pull request merged**